### PR TITLE
feat: implement ENV interceptor

### PR DESCRIPTION
**Pull request by waynexia**
# Which issue does this PR close?

Closes #

# Rationale for this change
 

In some cases, we may need to reference secret things in SQL, like the AWS token. This `ENV` interceptor provides a way to write sqlness cases with secret string.

# What changes are included in this PR?

Implement the `ENV` interceptor

# Are there any user-facing changes?

No
# How does this change test

Unit test and integration test

### CODE REVIEW

**Comment by jiacai2050** [comment]
@waynexia Ping, do you have free time to push this further? If not, I will continue based on your branch.

**Comment by waynexia** [comment]
> @waynexia Ping, do you have free time to push this further? If not, I will continue based on your branch.

Sorry for the delay. I'm not available in these few days. Is it okay to deliver it 3~5 days later?

**Comment by jiacai2050** [comment]
> > @waynexia Ping, do you have free time to push this further? If not, I will continue based on your branch.
> 
> Sorry for the delay. I'm not available in these few days. Is it okay to deliver it 3~5 days later?

Of course, take it easy.

**Comment by waynexia** [comment]
PTAL @jiacai2050

**Comment by jiacai2050** [comment]
@waynexia I will review this week.

File: sqlness/src/interceptor/env.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+/// -- SQLNESS ENV SECRET1 SECRET2
+/// SELECT $SECRET1, $SECRET2;
+///
+/// -- SQLNESS ENV SECRET1
+/// -- SQLNESS ENV SECRET2
+/// SELECT $SECRET1, $SECRET2;
+/// ````
+#[derive(Debug)]
+pub struct EnvInterceptor {
+    data: HashMap<String, String>,
```
Comment (jiacai2050) [inline sqlness/src/interceptor/env.rs:38]: Comments what's inside this map.

`data` is a general name, how about `env_vars`?

File: sqlness/src/interceptor/env.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+impl Interceptor for EnvInterceptor {
+    fn before_execute(
+        &self,
+        _: &mut Vec<String>,
```
Comment (jiacai2050) [inline sqlness/src/interceptor/env.rs]: It seems this args is not used in all interceptors, and I think iterceptor shouldn't change `display_query`.

**Review by jiacai2050 — APPROVED** [review]

**Pull request merged**